src/lib/storage/index/partial_hash/partial_hash_index_impl.cpp

```diff
@@ -0,0 +1,165 @@
+    return find_iter != indexed_chunks_to_unindex.cend();
+  };
+
+  // Iterate over all values stored in the index.
+  auto map_iter = _positions.begin();
+  while (map_iter != _positions.end()) {
+    auto& row_ids = _positions.at(map_iter->first);
+
+    // Remove every RowID entry of the value that references one of the chunks.
+    row_ids.erase(std::remove_if(row_ids.begin(), row_ids.end(), is_to_unindex), row_ids.end());
+
+    map_iter = row_ids.empty() ? _positions.erase(map_iter) : ++map_iter;
+  }
+
+  if (_null_positions.contains(DataType{})) {
+    auto& nulls = _null_positions[DataType{}];
+    nulls.erase(std::remove_if(nulls.begin(), nulls.end(), is_to_unindex), nulls.end());
+  }
+
+  return size_before - _indexed_chunk_ids.size();
+}
+
+template <typename DataType>
+bool PartialHashIndexImpl<DataType>::indexed_null_values() const {
+  return null_cbegin() != null_cend();
+}
+
+template <typename DataType>
+BasePartialHashIndexImpl::IteratorRange PartialHashIndexImpl<DataType>::range_equals(
+    const AllTypeVariant& value) const {
+  const auto begin = _positions.find(boost::get<DataType>(value));
+  if (begin == _positions.end()) {
+    const auto end_iter = cend();
+    return std::make_pair(end_iter, end_iter);
+  }
+  auto end = begin;
+  return std::make_pair(_create_iterator(begin), _create_iterator(++end));
+}
+
+template <typename DataType>
+BasePartialHashIndexImpl::IteratorRangePair PartialHashIndexImpl<DataType>::range_not_equals(
+    const AllTypeVariant& value) const {
+  const auto range_eq = range_equals(value);
+  return std::make_pair(std::make_pair(cbegin(), range_eq.first), std::make_pair(range_eq.second, cend()));
+}
+
+template <typename DataType>
+BasePartialHashIndexImpl::Iterator PartialHashIndexImpl<DataType>::cbegin() const {
+  return _create_iterator(_positions.cbegin());
+}
+
+template <typename DataType>
+BasePartialHashIndexImpl::Iterator PartialHashIndexImpl<DataType>::cend() const {
+  return _create_iterator(_positions.cend());
+}
+
+template <typename DataType>
+BasePartialHashIndexImpl::Iterator PartialHashIndexImpl<DataType>::null_cbegin() const {
+  return _create_iterator(_null_positions.cbegin());
+}
+
+template <typename DataType>
+BasePartialHashIndexImpl::Iterator PartialHashIndexImpl<DataType>::null_cend() const {
+  return _create_iterator(_null_positions.cend());
+}
+
+template <typename DataType>
+size_t PartialHashIndexImpl<DataType>::estimate_memory_usage() const {
+  auto bytes = size_t{0u};
+
+  bytes += sizeof(_indexed_chunk_ids);  // NOLINT - Linter complains size() should be used, which would be wrong.
+  bytes += sizeof(ChunkID) * _indexed_chunk_ids.size();
+
+  bytes += sizeof(_positions);
+  bytes += sizeof(DataType) * _positions.size();
+
+  bytes += sizeof(std::vector<RowID>) * _positions.size();
+  bytes += sizeof(RowID) * std::distance(cbegin(), cend());
+
+  bytes += sizeof(_null_positions);  // NOLINT
```

Comment:
see above

Reply:
Apparently the linter does not complain anymore.